### PR TITLE
Update pmem.yaml redisfailover example.

### PR DESCRIPTION
**Pull request by marcemq**
Changes proposed on the PR:
- Usage of upstreamed [`pmem/redis`](https://hub.docker.com/r/pmem/redis) image.
- Usage of `securityContext`.
- Separation of one argument by line into command statement.

### CODE REVIEW

**Comment by pohly** [comment]
Can you also explain *why* you are making these changes?

I know that running pmem-redis was broken. What was broken and which of these changes fixes that?

File: example/redisfailover/pmem.yaml [hunk 1]
```diff
@@ -15,14 +14,17 @@ spec:
...
-    image: redis_pmem # From https://github.com/pmem/redis
+    image: pmem/redis
     version: latest
     command:
       - "redis-server"
       - "/redis/redis.conf"
       - "--pmdir"
-      - "/data 100Mb"
+      - "/data"
+      - " 100Mb"
```
Comment (pohly) [inline example/redisfailover/pmem.yaml]: What is the value that `--pmdir` takes? A single string or two of them? Has that changed?

If it is now two, then you can remove the leading space in `  100Mb`. If it is one, then splitting that string into two parameters is not correct.

**Comment by pohly** [comment]
Now this makes sense to me, thanks.

Not sure what the commit message policy is for this repo (i.e. take this as a suggestion, not a requirement), but it would be good to add that information to the commit message and squash your two commits into one (because one is a fix for the other).

**Comment by ese** [comment]
thanks @marcemq for the PR. I agree with @pohly, good advice

**Comment by marcemq** [comment]
The new PR for this changes is https://github.com/spotahome/redis-operator/pull/169.
Sorry for the inconveniences.

**Pull request closed without merging**